test/e2e/support/utils/set-up-response-mocking.js

```diff
@@ -21,15 +31,23 @@
  * @param {Array} mocks Array of mock settings.
  */
 export async function setUpResponseMocking( mocks ) {
-	await page.setRequestInterception( true );
-	page.on( 'request', async ( request ) => {
-		for ( let i = 0; i < mocks.length; i++ ) {
-			const mock = mocks[ i ];
-			if ( mock.match( request ) ) {
-				await mock.onRequestMatch( request );
-				return;
+	if ( ! interceptionInitialized ) {
```

Comment:
What if two test files each want to set up their own response mocks? Wouldn't this singleton boolean global only allow the first set of mocks to "win"?